templates/commands/render/action_include.go

```diff
@@ -36,36 +36,32 @@ func actionInclude(ctx context.Context, inc *spec.Include, sp *stepParams) error
 }
 
 func includePath(ctx context.Context, inc *spec.IncludePath, sp *stepParams) error {
+	skipPaths, err := processPaths(inc.Skip, sp)
+	if err != nil {
+		return err
+	}
 	skip := make(map[string]struct{}, len(inc.Skip))
-	for _, s := range inc.Skip {
-		skipRelPath, err := parseAndExecuteGoTmpl(s.Pos, s.Val, sp.scope)
-		if err != nil {
-			return err
-		}
-		skip[skipRelPath] = struct{}{}
+	for _, s := range skipPaths {
+		skip[s.Val] = struct{}{}
 	}
 
-	for i, p := range inc.Paths {
-		// Paths may contain template expressions, so render them first.
-		walkRelPath, err := parseAndExecuteGoTmpl(p.Pos, p.Val, sp.scope)
-		if err != nil {
-			return err
-		}
+	incPaths, err := processPaths(inc.Paths, sp)
+	if err != nil {
+		return err
+	}
 
+	asPaths, err := processPaths(inc.As, sp)
+	if err != nil {
+		return err
+	}
+
+	for i, p := range incPaths {
 		// During validation in spec.go, we've already enforced that either:
 		//  - len(inc.As) == 0
 		//  - len(inc.As) == len(inc.Paths)
-		as := walkRelPath
+		as := p.Val
 		if len(inc.As) > 0 {
```

Comment:
Maybe len(asPaths) would be a bit readable than len(inc.As), even though they're the same length

Reply:
Good point, changed to asPaths.